From 69cb9b1f36f99ff64420fd9e1bcc5bc7b3ec585f Mon Sep 17 00:00:00 2001
From: Qiang Yu <yuq825@gmail.com>
Date: Mon, 4 Sep 2017 17:51:04 +0800
Subject: [PATCH 078/134] drm/lima: fix deprecated
 drm_calloc_large/drm_free_large

v2:
  (Qiang Yu)
  use kvzalloc instead of kvmalloc_array

Signed-off-by: Erico Nunes <nunes.erico@gmail.com>
Signed-off-by: Qiang Yu <yuq825@gmail.com>
---
 drivers/gpu/drm/lima/lima_vm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/lima/lima_vm.c b/drivers/gpu/drm/lima/lima_vm.c
index 008391e..29ca33b 100644
--- a/drivers/gpu/drm/lima/lima_vm.c
+++ b/drivers/gpu/drm/lima/lima_vm.c
@@ -159,7 +159,7 @@ struct lima_vm *lima_vm_create(struct lima_device *dev)
 {
 	struct lima_vm *vm;
 
-	vm = drm_calloc_large(1, sizeof(*vm));
+	vm = kvzalloc(sizeof(*vm), GFP_KERNEL);
 	if (!vm)
 		return NULL;
 
@@ -176,7 +176,7 @@ struct lima_vm *lima_vm_create(struct lima_device *dev)
 	return vm;
 
 err_out:
-	drm_free_large(vm);
+	kvfree(vm);
 	return NULL;
 }
 
@@ -216,7 +216,7 @@ void lima_vm_release(struct kref *kref)
 		dma_free_coherent(vm->dev->dev, LIMA_PAGE_SIZE, vm->pd.cpu,
 				  vm->pd.dma & ~LIMA_PAGE_MASK);
 
-	drm_free_large(vm);
+	kvfree(vm);
 }
 
 void lima_vm_print(struct lima_vm *vm)
-- 
2.0.1

